toolchain/check/import_cpp.cpp

```diff
@@ -446,18 +485,25 @@ static auto ImportFunctionDecl(Context& context, SemIR::LocId loc_id,
     context.TODO(loc_id, "Unsupported: Template function");
     return SemIR::ErrorInst::InstId;
   }
-  auto param_patterns_id =
-      MakeParamPatternsBlockId(context, loc_id, *clang_decl);
-  if (!param_patterns_id.has_value()) {
-    return SemIR::ErrorInst::InstId;
-  }
-  auto return_slot_pattern_id = GetReturnType(context, loc_id, clang_decl);
-  if (SemIR::ErrorInst::InstId == return_slot_pattern_id) {
+
+  context.inst_block_stack().Push();
+  context.pattern_block_stack().Push();
+
+  auto param_patterns_id = SemIR::InstBlockId::Empty;
+  auto return_slot_pattern_id = SemIR::InstId::None;
+  auto call_params_id = SemIR::InstBlockId::Empty;
+  auto success =
+      GetFunctionParams(context, loc_id, clang_decl, param_patterns_id,
+                        return_slot_pattern_id, call_params_id);
+
+  auto decl_block_id = context.inst_block_stack().Pop();
+  auto pattern_block_id = context.pattern_block_stack().Pop();
```

Comment:
nit: I think popping `pattern_block_stack` before `inst_block_stack` to reverse the order you pushed it makes sense, though it definitely doesn't really matter as long as you don't have any code in between.